gateway/modules/auth/match_test.go

```diff
@@ -334,19 +334,34 @@ func TestModule_matchEncrypt(t *testing.T) {
 			name: "valid res",
 			m:    &Module{aesKey: base64DecodeString("Olw6AhA/GzSxfhwKLxO7JJsUL6VUwwGEFTgxzoZPy9g=")},
 			args: args{rule: &config.Rule{Rule: "encrypt", Fields: []string{"res.username"}}, args: map[string]interface{}{"res": map[string]interface{}{"username": "username1"}}},
-			want: &model.PostProcess{PostProcessAction: []model.PostProcessAction{model.PostProcessAction{Action: "encrypt", Field: "res.username"}}},
+			want: &model.PostProcess{PostProcessAction: []model.PostProcessAction{{Action: "encrypt", Field: "res.username"}}},
 		},
 		{
 			name:    "invalid field prefix",
 			m:       &Module{aesKey: base64DecodeString("Olw6AhA/GzSxfhwKLxO7JJsUL6VUwwGEFTgxzoZPy9g=")},
 			args:    args{rule: &config.Rule{Rule: "encrypt", Fields: []string{"abc.username"}}, args: map[string]interface{}{"abc": map[string]interface{}{"username": "username1"}}},
 			wantErr: true,
 		},
+		{
+			name: "rule clause - allow",
+			m:    &Module{project: "project"},
+			args: args{
+				rule: &config.Rule{Rule: "encrypt", Clause: &config.Rule{Rule: "allow"}},
+			},
+			want: &model.PostProcess{},
+		},
+		{
+			name: "rule clause - deny",
+			m:    &Module{project: "project"},
+			args: args{
+				rule: &config.Rule{Rule: "encrypt", Clause: &config.Rule{Rule: "deny"}},
+			},
+		},
 	}
 
 	for _, tt := range tests {
```

Comment:
You think we should check if the fields are getting encrypted or not? A simple check to see whether its changed or not can be helpful

Reply:
Sure, on it!